pages/templates/_slug.vue

```diff
@@ -0,0 +1,166 @@
+<template>
+  <div>
+    <div v-if="template" class="template-preview-container">
+      <div class="preview-heading-wrapper">
+        <NuxtLink :to="`/templates/`" class="d-flex align-items-center">
+          <img
+            class="template__back"
+            src="@/assets/images/back-arrow.svg"
+            alt="back-button"
+          />
+          <span class="preview-heading">
+            {{ template.name }}
+          </span>
+        </NuxtLink>
+        <button
+          class="preview-use-template-btn"
+          @click="redirectTo(template.id)"
+        >
+          Use Template
+        </button>
+      </div>
+
+      <iframe
+        :src="template.surveyUrl"
+        frameborder="0"
+        width="100%"
+        class="template-preview"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+// MetaTags
+import getSiteMeta from '../../utils/getSiteMeta'
+
+export default {
+  layout: 'message',
+  data() {
+    return {
+      template: {},
+    }
+  },
+  mounted() {
+    this.getTemplate()
+  },
+  computed: {
+    meta() {
+      const metaData = {
+        type: 'website',
+        url: `https://formester.com/templates/${this.$route.params.slug}/`,
+        title: this.template?.name || 'Formester',
+        description: this.template?.description,
```

Comment:
should also have a default description which would work with the title incase the description is not coming